.github/workflows/release.yml

```diff
@@ -0,0 +1,37 @@
+name: Build and Release
+on:
+  # Runs automatically when a tag beginning with 'release-' is pushed.
+  push:
+    tags:
+      - release-*
+
+jobs:
+  gh-release:
+    runs-on: ubuntu-20.04
+    env:
+      BOULDER_TOOLS_TAG: go1.17.7_2022-02-10
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: dockerhub login
+        uses: docker/login-action@v1.8.0
+        with:
+          username: ${{ secrets.DOCKER_USERNAME}}
+          password: ${{ secrets.DOCKER_PASSWORD}}
+          logout: true
+        continue-on-error: true
+
+      - name: docker-compose pull
+        run: docker-compose pull
+
+      - name: build and release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        # https://cli.github.com/manual/gh_release_create
+        run: |
+          docker-compose run boulder ./make-deb.sh
```

Comment:
Can this command populate ARCHIVEDIR and run `make deb` directly, rather than relying on a two-line script? (I assume the answer is no, otherwise you wouldn't have done it this way, but figured I'd ask just in case there is a way to prevent the creation of a new single-use bash file.)

Reply:
Every permutation I attempted led to failure.

```shell
docker-compose run boulder ./make-deb.sh "ARCHIVEDIR=$PWD && make deb"
docker-compose run boulder ./make-deb.sh sh -c "ARCHIVEDIR=$PWD && make deb"
docker-compose run boulder ./make-deb.sh bash -c "ARCHIVEDIR=$PWD && make deb"
docker-compose run boulder ./make-deb.sh 'sh -c "ARCHIVEDIR=$PWD && make deb"'
docker-compose run boulder ./make-deb.sh 'bash -c "ARCHIVEDIR=$PWD && make deb"'
```
If you've got some other ideas for me to try, please let me know.

Reply:
I would do something like:

```
docker-compose run boulder ARCHIVEDIR=$PWD make deb
docker-compose run boulder make ARCHIVEDIR=$PWD  
```

The former asks bash to set an environment variable for a single command. The latter asks make to set one of its internal variables.

Reply:
Although both of these suffer from the problem that PWD is evaluated _outside_ the docker container. So maybe:

```
docker-compose run boulder ARCHIVEDIR=. make deb
docker-compose run boulder make ARCHIVEDIR=. deb
```

Reply:
Resolving as we decided to perform the build outside of docker.